### PR TITLE
[DataFactory] Add Chaining Trigger

### PR DESCRIPTION
**Pull request by haricr**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 1]
```diff
@@ -4529,6 +4537,22 @@
             "type": "string"
           },
           "readOnly": true
+        },
+        "runDimension": {
+          "description": "Run dimention for which trigger was fired.",
```
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]:
```suggestion
          "description": "Run dimension for which trigger was fired.",
```

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Trigger.json [hunk 2]
```diff
@@ -585,6 +585,50 @@
...
+                "$ref": "../datafactory.json#/definitions/PipelineReference"
+              },
+              "description": "Dependent Pipelines."
+            },
+            "retryPolicy": {
+              "$ref": "#/definitions/RetryPolicy",
+              "description": "Retry policy that will be applied for failed pipeline runs."
+            },
+            "runDimension": {
+              "description": "Run Dimenstion property that needs to be emitted by dependent pipelines.",
```
Comment (nschonni) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Trigger.json]:
```suggestion
              "description": "Run Dimension property that needs to be emitted by dependent pipelines.",
```

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 3]
```diff
@@ -41,6 +41,14 @@
             "type": "object"
           }
         },
+        "runDimensions": {
+          "description": "Dimensions emitted by Pipeline.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "object",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: How come this one is object, the ones in the API are strings?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Trigger.json [hunk 4]
```diff
@@ -585,6 +585,50 @@
...
+          "$ref": "#/definitions/Trigger"
+        }
+      ],
+      "properties": {
+        "pipeline": {
+          "$ref": "../datafactory.json#/definitions/TriggerPipelineReference",
+          "description": "Pipeline for which runs are created when all dependent pipelines complete successfully."
+        },
+        "typeProperties": {
+          "description": "Chaining Trigger properties.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Trigger.json]: How come this is not a separate type as all other types in our swagger

**Comment by michaeljqzq** [comment]
please take a look at CI task logs

**Comment by michaeljqzq** [comment]
@haricr , is this PR still valid? If not, I will close it

**Pull request closed without merging**